src/View.php

```diff
@@ -613,16 +613,38 @@ public function jsUrl($page = [])
      */
     public function url($page = [])
     {
-        return $this->getApp()->url($page, false, $this->_getStickyArgs());
+        return $this->getApp()->url($page, false, array_merge($this->getRunningCallbackArgs(false, is_array($page) ? $page : []), $this->getStickyArgs()));
+    }
+
+    protected function getRunningCallbackArgs(bool $isTerminated, array $page): array
+    {
+        $args = [];
+        foreach ($this->elements as $v) {
+            if ($v instanceof Callback) { // @phpstan-ignore-line
+                if (($page[Callback::URL_QUERY_TARGET] ?? null) === $v->getUrlTrigger()) {
```

Comment:
@ibelar can you review and look at the Behat testing?

1. callback triggers are now based purely on render tree
2. the `CallbackTest::testViewUrlCallback` should cover the original Behat testing
3. Behat testing is failing, not sure exactly why, maybe some extra condition like in https://github.com/atk4/ui/commit/e3c299b36cc26bc30eef3360d3f0aa188c1a53dc#diff-c5bb4a733c5ddd3bb0744ef4c29a8fd1e0bd5999fcf06b7f3cfef4c3d1926ce3R624 is needed - but please make sure, point 1 stays true, eg. $_GET is never evaluated

Reply:
It seems that the delete action callback (JsCallback trigger argument) is added to the add action callback URL.
That is why you get the exception. It tries to run (`JsCallbakc::set()`) the delete action first.

<img width="625" alt="Screen Shot 2021-07-26 at 12 04 16 PM" src="https://user-images.githubusercontent.com/2204478/127021744-66333d5d-ed2a-48bf-a678-a21eb66dce8f.png">

Reply:
any idea what is wrong with the currect algorithm?

Consider this render tree:
```
App -> V1 -> C1
          -> C2
          -> V2 -> C3
```

currently, when URL for C3 is rendered, both C1 and C2 URL triggers are added. Is that correct?